src/app/server/CommissioningWindowManager.cpp

```diff
@@ -85,6 +91,7 @@ void CommissioningWindowManager::Cleanup()
 
 void CommissioningWindowManager::OnSessionEstablishmentError(CHIP_ERROR err)
 {
+    DeviceLayer::SystemLayer().CancelTimer(HandleSessionEstablishmentTimeout, this);
```

Comment:
I was wondering if we should allow access to SystemLayer from App directly in future. App should interact with the systemLayer indirectly such as from mServer->GetExchangeManager()

Reply:
Makes sense for apps. `CommissioningWindowManager.cpp` falls somewhere in the middle though. It's some functionality of server that is split into its own class.